RHDEVDOCS-4477 update concepts for odo 3.0 (#51265)
openshift/openshift-docs: modules/odo-core-concepts.adoc

`odo` abstracts Kubernetes concepts into terminology that is familiar to developers:

Application::

> A typical application, developed using a link:https://www.redhat.com/en/topics/cloud-native-apps[cloud-native approach], that performs a particular task.

Reviewer: ```suggestion
A typical application, developed using a link:https://www.redhat.com/en/topics/cloud-native-apps[cloud-native approach], that does a particular task.
```
Per the ISG, `perform` should be avoided if possible.

Author: "perform verb
Do not use if more direct wording is possible. For example, instead of writing "Perform the product installation", write "Install the product"."

I think perform is ok here, because it needs some sort of verb like "execute", or "does" or whatever, so I don't think any of those are more direct in this instance.

+
Examples of _applications_ include online video streaming, online shopping, and hotel reservation systems.

Component::
A set of Kubernetes resources that can run and be deployed separately. A cloud-native application is a collection of small, independent, loosely coupled _components_. `odo` uses Devfiles for the definition of _components_.
+
Examples of _components_ include an API back-end, a web interface, and a payment back-end.

Project::

> A _project_ helps achieve multi-tenancy, allowing different teams to deploy multiple applications on the cluster. In Kubernetes, a _project_ is represented by a namespace.

Reviewer: ```suggestion
A single collection of code, tests, and libraries that helps achieve multi-tenancy so that different teams can deploy multiple applications on a cluster. In Kubernetes, a _project_ is represented by a namespace.
```
Currently, the first sentence is structured as a complete sentence and includes the term being defined, unlike the other definitions in the list. My suggestion is one possible way to make this sentence parallel in structure with the other definitions and to avoid using `allowing`, which we can't use, per ISG guidelines.

Author: I think projects have changed from 2.5, so they are no longer that list of items, but now just match a `Project` in OPenShift and a `Namespace` in Kubernetes, so I have put:

A _project_ helps achieve multi-tenancy, so that different teams can deploy multiple applications on the cluster. In Kubernetes, a _project_ is represented by a `Namespace` resource, while in OpenShift it is a `Project` resource.

Context::
A directory that contains the source code, tests, libraries, and `odo` configuration files for a single _component_.

Devfile::

> An open standard for defining containerized development environments, using portable YAML files that contain the definition for a component and its parts. Devfiles enable developer tools to simplify and speed up workflows. For more information, see the documentation at link:https://devfile.io[].

Reviewer: ```suggestion
An open standard for defining containerized development environments. A Devfile uses portable YAML files that contain the definition for a component and its parts. Devfiles enable developer tools to simplify and speed up workflows. For more information, see the documentation at link:https://devfile.io[].
```

Author: Implemented your suggestion as follows:

"...containerized development environments. Devfiles are portable YAML files..."
